core/jobs/batch_jobs/exp_validation_jobs.py

```diff
@@ -0,0 +1,828 @@
+            | 'Show info for state interaction part 2' >> beam.Map(
+                lambda objects: job_run_result.JobRunResult.as_stderr(
+                   f'The id of exp is {objects[0]}, ' +
+                   f'created on {objects[2]}, and the state interaction'
+                   + f' part 2 erroneous data are {objects[1]}'
+                )
+            )
+        )
+
+        invalid_exps_with_errored_state_interaction_values_part_3 = (
+            combine_exp_ids_and_states
+            | 'Get invalid state interaction values part 3' >> beam.Map(
+                lambda objects: (
+                    objects[0],
+                    self.filter_invalid_state_interaction_values_part_3(
+                        objects[1]), objects[2]))
+            | 'Remove empty values for state interaction part 3' >> beam.Map(
+                lambda objects: (
+                    objects[0], self.remove_empty_values(objects[1]),
+                    objects[2].date()))
+        )
+
+        report_invalid_state_interaction_values_part_3 = (
+            invalid_exps_with_errored_state_interaction_values_part_3
+            | 'Show info for state interaction part 3' >> beam.Map(
+                lambda objects: job_run_result.JobRunResult.as_stderr(
+                   f'The id of exp is {objects[0]}, ' +
+                   f'created on {objects[2]}, and the state interaction'
+                   + f' part 3 erroneous data are {objects[1]}'
+                )
+            )
+        )
+
+        invalid_exps_with_errored_states_values = (
+            combine_exp_ids_and_states
+            | 'Get invalid state values' >> beam.Map(
+                lambda objects: (
+                    objects[0],
+                    self.filter_invalid_state_values(objects[1]), objects[2]))
+            | 'Remove empty values for state values' >> beam.Map(
+                lambda objects: (
+                    objects[0], self.remove_empty_values(objects[1]),
+                    objects[2].date()))
+        )
+
+        report_invalid_state_values = (
+            invalid_exps_with_errored_states_values
+            | 'Show info for state values' >> beam.Map(
+                lambda objects: job_run_result.JobRunResult.as_stderr(
+                   f'The id of exp is {objects[0]}, ' +
+                   f'created on {objects[2]}, and the state' +
+                   f' erroneous data are {objects[1]}'
+                )
+            )
+        )
+
+        return (
+            (
+                report_number_of_exps_queried,
+                report_invalid_state_rte_values,
+                report_invalid_state_interaction_values_part_1,
+                report_invalid_state_interaction_values_part_2,
+                report_invalid_state_interaction_values_part_3,
+                report_invalid_state_values
+            )
+            | 'Combine results' >> beam.Flatten()
+        )
+
+    def filter_invalid_state_rte_values(
```

Comment:
Move these methods above the run functions. And also probably use static methods.

Reply:
Done